frontend/src/navigation/routing/routing.tsx

```diff
@@ -12,23 +12,23 @@ import { ResetPassword } from '@components/reset-password/reset-password';
 import { SearchPage } from '@components/search-page/search-page';
 import { RedirectAfterSign } from '@components/sign/components/redirect-after-sign/redirect-after-sign';
 import { Sign } from '@components/sign/sign';
+import { useAppSelector } from '@hooks/hooks';
 import { ProtectedRoute } from '@navigation/protected-route/protected-route';
+import { useGetUserQuery } from '@store/queries/user/update-user';
 
 const Routing: FC = () => {
-  const authData = {
-    name: 'Oleksandr',
-    role: 'admin',
-  };
-  const { role } = authData;
-  const isAdmin = role === 'admin';
+  const { data: authData } = useGetUserQuery();
+  const isAdmin = authData?.role === 'admin';
```

Comment:
I'm afraid that this won't work as expected in the scenario when the admin page is reloaded or visited directly by the link :sweat_smile:. Because on the first render the `authData` is undefined and the `isAdmin` will never be `true`. Therefore, the `<Route element={<ProtectedRoute isAllowed={isAdmin} />}>` will redirect a probable admin to the sign-in page.

Reply:
@oleksandr-danylchenko Yes, you are absolutely right. I would appreciate some ideas about other way to send user role request at router level. Right now I don't see any other options if we can't use local storage and redux state

Reply:
I think we can use `jwt-decode` package to decode jwt on client so this way we will know role on the first render. The only problem is that we can't check if token is valid. But if it's not we will get an error after fetching and will be redireted from protected page.

Just a suggestion.

Reply:
@R0ixy 
We mustn't get the role of the the user from token stored in the local storage. Because that token can be easily modified on the client side. Also, if the role was changed from the admin panel, the token won't know about it

Reply:
Why won't we just wait till the user will be loaded and only then check if they are an admin? :eyes: 
And before the user will be loaded we can show a global loaded